spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcp/FunctionInvoker.java

```diff
@@ -98,6 +98,11 @@ public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws E
 					httpResponse.appendHeader(header.getKey(), header.getValue().toString());
 				}
 			}
+			httpResponse.setStatusCode(200);
+		}
+		catch (Exception e) {
+			log.error("Error in HTTP Function Invoker", e);
+			httpResponse.setStatusCode(500, e.getMessage());
```

Comment:
might want to consider not propagating the actual message, since the actual message may have sensetive data.

Reply:
Thanks Ray, done. I'll also look into trying to find where that error is being caught by the framework.